qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -226,47 +252,35 @@ def _draw_widgets(self):
 
         # Read the list and set widgets accordingly
         for segment in config.val.statusbar.widgets:
-            if segment == 'url':
-                self._hbox.addWidget(self.url)
-                self.url.show()
-            elif segment == 'scroll':
-                self._hbox.addWidget(self.percentage)
-                self.percentage.show()
-            elif segment == 'scroll_raw':
-                self._hbox.addWidget(self.percentage)
-                self.percentage.set_raw()
-                self.percentage.show()
-            elif segment == 'history':
-                self._hbox.addWidget(self.backforward)
-                self.backforward.enabled = True
-                if tab:
-                    self.backforward.on_tab_changed(tab)
-            elif segment == 'tabs':
-                self._hbox.addWidget(self.tabindex)
-                self.tabindex.show()
-            elif segment == 'keypress':
-                self._hbox.addWidget(self.keystring)
-                self.keystring.show()
-            elif segment == 'progress':
-                self._hbox.addWidget(self.prog)
-                self.prog.enabled = True
+            cur_widget = self._get_widget_from_config(segment)
+            self._hbox.addWidget(cur_widget)
+
+            if segment == 'scroll_raw':
+                cur_widget.set_raw()
+            elif segment in ('history', 'progress'):
+                cur_widget.enabled = True
                 if tab:
-                    self.prog.on_tab_changed(tab)
+                    cur_widget.on_tab_changed(tab)
+
+                # Do not call .show() for these widgets.
+                return
```

Comment:
It seems like removing this return fixes it for me. This is confusing, since in the past we did not call `show` on the history and progress widget, which is why I explicitly inserted this return. Looking at it now it seems a bit weird that they don't have it, but it worked before, right?

Reply:
Ah, this also has the effect that I get a full progress bar that stays when freshly setting this setting. It vanishes after reloading or even switching tabs back and forth.

Reply:
Yup - those widgets should not be shown by default, as other parts of the code hide/show them as needed - at least with the progress bar, not sure off-hand why `history` wouldn't be shown!

You were on the right track, however: This should be a `continue`, not a `return`! Otherwise, we abort the `for` loop we are in entirely, and any widgets after `history` will not be added at all.

Fixed in ea140b0333b9c3d24a939516fd99a58543e75342. Looks like it's all done for this PR then!